src/crates_cache.rs

```diff
@@ -179,6 +179,7 @@ impl CratesCache {
         let reader = bar.wrap_read(response.into_reader());
         let ungzip = GzDecoder::new(reader);
         let mut archive = tar::Archive::new(ungzip);
+        let mut meta: Option<Metadata> = None;
```

Comment:
Here we'll need to invalidate the current `meta` file because the different `store` calls below might fail mid-way and if they do they've invalidate their respective data files. Another strategy would be to download the files into adjacent temporary files—adjacent being relevant so that `fs::rename` does not need to move across file systems.